source/module_io/input.cpp

```diff
@@ -4223,6 +4231,36 @@ void Input::strtolower(char* sa, char* sb)
     sb[len] = '\0';
 }
 
+template <typename T>
+void Input::read_value2stdvector(std::ifstream& ifs, std::vector<T>& var)
+{
+    // reset var
+    var.clear(); var.shrink_to_fit();
+    std::string line;
+    std::getline(ifs, line); // read the whole rest of line
+    std::vector<char> temp;
+    for(char &c: line)
+    {
+        if(c == '\t' || c == ' ' || c == '\n' || c == '#') // space or tab seperates values
+        {
+            if(temp.size() > 0) // if temp is not empty, excludes the case of multiple spaces or tabs
+            {
+                std::string str(temp.begin(), temp.end());
+                var.push_back(std::stod(str));
+                temp.clear();
+            }
+            if(c == '\n' || c == '#' || c == '\0') break; // end of line
+        }
+        else temp.push_back(c); // other characters
+    }
+    if(temp.size() > 0) // the last value
+    {
+        std::string str(temp.begin(), temp.end());
+        var.push_back(std::stod(str));
```

Comment:
`std::stod` will convert `string` into `double` instead of template `T`.

Reply:
thanks, I have used enable_if to properly cast std::string